scripts/configure.utils.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2023 Project CHIP Authors
```

Comment:
rename this to not be utils/support/helper/tools etc. Otherwise it becomes a grab-all of everything in it (is it already? that is not good for maintenance/muscle memory)

Reply:
I've renamed the file to `configure.impl.py` to better convey the fact that it's an implementation detail.